url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/graph/EdgeRenderer.java

```diff
@@ -0,0 +1,1109 @@
+				poly.addPoint((int) dst.getX(), (int) dst.getY());
+			} else
+				dst.setLocation((int) (dst.getX() + ax),
+						(int) (dst.getY() + ay));
+			poly.addPoint((int) (last.getX() + ax - ay / 2), (int) (last.getY()
+					+ ay + ax / 2));
+			return poly;
+
+		} else if (style == GraphConstants.ARROW_SIMPLE) {
+			GeneralPath path = new GeneralPath(GeneralPath.WIND_NON_ZERO, 4);
+			path.moveTo((float) (dst.getX() + ax + ay / 2), (float) (dst.getY()
+					+ ay - ax / 2));
+			path.lineTo((float) dst.getX(), (float) dst.getY());
+			path.lineTo((float) (dst.getX() + ax - ay / 2), (float) (dst.getY()
+					+ ay + ax / 2));
+			return path;
+
+		} else if (style == GraphConstants.ARROW_CIRCLE) {
+			Ellipse2D ellipse = new Ellipse2D.Float((float) (dst.getX() + ax
+					/ 2 - size / 2), (float) (dst.getY() + ay / 2 - size / 2),
+					size, size);
+			dst.setLocation(dst.getX() + ax, dst.getY() + ay);
+			return ellipse;
+
+		} else if (style == GraphConstants.ARROW_LINE
+				|| style == GraphConstants.ARROW_DOUBLELINE) {
+			GeneralPath path = new GeneralPath(GeneralPath.WIND_NON_ZERO, 4);
+			path.moveTo((float) (dst.getX() + ax / 2 + ay / 2), (float) (dst
+					.getY()
+					+ ay / 2 - ax / 2));
+			path.lineTo((float) (dst.getX() + ax / 2 - ay / 2), (float) (dst
+					.getY()
+					+ ay / 2 + ax / 2));
+			if (style == GraphConstants.ARROW_DOUBLELINE) {
+				path.moveTo((float) (dst.getX() + ax / 3 + ay / 2),
+						(float) (dst.getY() + ay / 3 - ax / 2));
+				path.lineTo((float) (dst.getX() + ax / 3 - ay / 2),
+						(float) (dst.getY() + ay / 3 + ax / 2));
+			}
+			return path;
+		}
+		return null;
+	}
+
+	/**
+	 * @return Returns the gradientColor.
+	 */
+	public Color getGradientColor() {
+		return gradientColor;
+	}
+
+	/**
+	 * @param gradientColor
+	 *            The gradientColor to set.
+	 */
+	public void setGradientColor(Color gradientColor) {
+		this.gradientColor = gradientColor;
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note </a> for more information.
+	 */
+	public void validate() {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note </a> for more information.
+	 */
+	public void revalidate() {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note </a> for more information.
+	 */
+	public void repaint(long tm, int x, int y, int width, int height) {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note </a> for more information.
+	 */
+	public void repaint(Rectangle r) {
+	}
+
+	/**
+	 * Overridden for performance reasons. See the <a
+	 * href="#override">Implementation Note </a> for more information.
+	 */
+	protected void firePropertyChange(String propertyName, Object oldValue,
+			Object newValue) {
+		// Strings get interned...
+		if (propertyName == "text")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.